handbook/engineering/code-intelligence/goals.md

```diff
@@ -0,0 +1,45 @@
+# Cloud goals and priorities
+
+## Goals
+
+Track our progress in our [Looker dashboard](https://sourcegraph.looker.com/dashboards/131).
+
+### The majority of code intel results are precise (for languages we support fully)
+
+_Languages we fully support currently include Go, TypeScript, C/C++, and Java._ Languages will be added to this list as we make progress toward our second goal, listed below.
+
+It is imperative that we raise adoption *deeply* within a particular language and ecosystem. The goal of the code intelligence team is to provide fast and precise answers about a user's _universe of code_. This can only happen when that entire universe is indexed. We intend to provide an index for _all_ such universes of code by making progress on two fronts. First, we need to raise excitement for the opportunities adopting precise code intelligence makes possible. Once a user sees what is possible, they should have little reason not to put in the effort of adoption. Second, we need to reduce the friction felt during indexing (especially during adoption). Users tend to perceive only the work we ask them to do to set up precise code intelligence and do not perceive the full benefit of precise code intelligence. Reducing the friction of indexing is a critical step in growing support.
```

Comment:
This is a great paragraph on what we need to do (and we should keep it), I'd also love to see an added paragraph about the value this brings to the user - the problem that we solve for them. Why would I, the user, care about making LSIF configuration easier? I don't care if I don't understand why precise code intel will change my life as a developer.

Reply:
Added a short Problem section to address this.

```suggestion
**Problem:** By default, users get basic code intel results which are often incorrect for tokens with common names (such as Get). This is solved by enabling precise code intel, which gives users faster, 100% accurate results for a definition or reference for a symbol they hover over. 

It is imperative that we raise adoption *deeply* within a particular language and ecosystem. We want to provide fast and precise answers about a user's _universe of code_. This can only happen when that entire universe is indexed. We intend to provide an index for _all_ such universes of code by making progress on two fronts. First, we need to raise excitement for the opportunities adopting precise code intelligence makes possible. Once a user sees what is possible, they should have little reason not to put in the effort of adoption. Second, we need to reduce the friction felt during indexing (especially during adoption). Users tend to perceive only the work we ask them to do to set up precise code intelligence and do not perceive the full benefit of precise code intelligence. Reducing the friction of indexing is a critical step in growing support.
```